driver-core/src/main/java/com/datastax/driver/core/policies/RetryPolicy.java

```diff
@@ -50,10 +50,17 @@
 
         private final Type type;
         private final ConsistencyLevel retryCL;
+        private final boolean retryCurrent;
 
+        private RetryDecision(Type type, ConsistencyLevel retryCL, boolean retryCurrent){
+            this.type = type;
+            this.retryCL = retryCL;
+            this.retryCurrent = retryCurrent;
+        }
         private RetryDecision(Type type, ConsistencyLevel retryCL) {
```

Comment:
This constructor is only called 3 times and from inside the class. We can remove it and call the other one with retryCurrent explicitly set to true.